docs/frontend-system/architecture/20-extensions.md

```diff
@@ -293,48 +280,68 @@ app:
+
+```tsx
+const exampleExtension = createExtension({
+  // ...
+  config: {
+    schema: {
+      disableIcon: z.boolean().default(false),
+    },
+  },
+  output: [coreExtensionData.reactElement, iconDataRef.optional()],
+  *factory({ config }) {
+    yield coreExtensionData.reactElement(<div>Hello World</div>);
+
+    if (!config.disableIcon) {
+      yield iconDataRef(<ExampleIcon />);
+    }
+  },
+});
+```
+
+## Extension boundary
 
 The `ExtensionBoundary` wraps extensions with several React contexts for different purposes
 
 ### Suspense
 
-All React elements rendered by extension creators should be wrapped in the extension boundary. With `Suspense` the extension can than load resources asynchronously with having a loading fallback. It also allows to lazy load the whole extension similar to how plugins are currently lazy loaded in Backstage.
+Most React elements rendered by extensions should be wrapped in the extension boundary. With `Suspense` the extension can than load resources asynchronously with having a loading fallback. It also allows to lazy load the whole extension similar to how plugins are currently lazy loaded in Backstage.
 
-### Error Boundary
+### Error boundary
 
 Similar to plugins the `ErrorBoundary` for extension allows to pass in a fallback component in case there is an uncaught error inside of the component. With this the error can be isolated & it would prevent the rest of the plugin to crash.
 
 ### Analytics
 
 Analytics information are provided through the `AnalyticsContext`, which will give `extensionId` & `pluginId` as context to analytics event fired inside of the extension. Additionally `RouteTracker` will capture an analytics event for routable extension to inform which extension metadata gets associated with a navigation event when the route navigated to is a gathered `mountPoint`. Whether an extension is routable is inferred from its outputs, but you can also explicitly control this behavior by passing the `routable` prop to `ExtensionBoundary`.
 
-The `ExtensionBoundary` can be used like the following in an extension creator:
+The `ExtensionBoundary` can be used like the following in an extension:
 
 ```tsx
-export function createSomeExtension<
-  TConfig extends {},
-  TInputs extends AnyExtensionInputMap,
->(options): ExtensionDefinition<TConfig> {
-  return createExtension({
-    // ...
-    factory({ config, inputs, node }) {
-      const ExtensionComponent = lazy(() =>
-        options
-          .loader({ config, inputs })
-          .then(element => ({ default: () => element })),
-      );
-
-      return {
-        path: config.path,
-        routeRef: options.routeRef,
-        element: (
-          <ExtensionBoundary node={node}>
-            <ExtensionComponent />
-          </ExtensionBoundary>
-        ),
-      };
-    },
-  });
-}
+const routableExtension = createExtension({
+  // ...
+  factory({ config, inputs, node }) {
+    const ExtensionComponent = lazy(() =>
+      options
```

Comment:
Unclear where this options comes from - this was probably taken from an old extension creator right?

Reply:
Yep, needs an update